go/private/context.bzl

```diff
@@ -267,13 +270,17 @@ def _go_context_data(ctx):
   linker_options = [o for o in raw_linker_options if not o in [
     "-Wl,--gc-sections",
   ]]
+  env = {}
```

Comment:
What would go here in the future? We generally want to avoid pulling anything out of the user's environment, since that makes builds less reproducible.

Reply:
this is not about pulling the user's environment, but rather merging the rule environment with the builder's or `go install` based rules such as `stdlib`)

Reply:
Ok, that makes sense.